### PR TITLE
The volume slider from immersive mode remains displayed even after mo…

### PR DESCRIPTION
**Pull request by daron-walters**
…ving the pointer away - #2948

1. updated code to use the x location of the volume button as the starting point for hover to show/hide slider.  Per design doc the slider should show from the beginning of the volume button to the end of the slider

2. added code to handle hide the slider when the x location is outside of the media controls.  On oculus quest the x coord is always negetive when pointer is outside control area

### CODE REVIEW

File: app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/MediaControlsWidget.java [hunk 1]
```diff
@@ -197,8 +197,13 @@ public void onSeekPreview(String aText, double aRatio) {
             if (mMedia == null) {
                 return false;
             }
+            if(event.getX()<0)
+            {
+                mVolumeControl.setVisibility(View.GONE);
+                return false;
+            }
```
Comment (bluemarvin) [inline app/src/common/shared/org/mozilla/vrbrowser/ui/widgets/MediaControlsWidget.java:204]: A few formatting nits:
```
if(event.getX()<0)
{
}
```
Should be:
```
if (event.getX() < 0) {
}
```
Space after `if`. Space on each side of `<`.  Curly brackets `{}` should be K&R style.

**Comment by bluemarvin** [comment]
@keianhzo I sort of wonder if the better solution is to show the volume bar when the speaker button is hovered and then start a timer after the pointer leaves the volume button that hides the volume slider if the pointer does not enter the volume slider after a certain time. If the pointer does enter the volume slider, the timer should be reset and started again once the pointer leaves the slider. Thoughts?

**Comment by keianhzo** [comment]
@bluemarvin @daron-walters That's a good option. 
Other options:
- We show the volume slider when the pointer leaves the button through the right side of the volume button. In any other case we hide it. We can do this handling the threshold both the widget hover listener and in the button hover listener.
- Or we just always show the volume bar, I don't see any good reason (other than the spec) why it shouldn't be always visible and we save the user all this hovering thing.

**Comment by daron-walters** [comment]
All three are good options, I will default to you guys on the final decision.  Just let me know and I can start on implementation

**Comment by keianhzo** [comment]
@daron-walters I think @bluemarvin idea is probably the most simple-effective one, I'd go ahead with that one.

**Comment by daron-walters** [comment]
@keianhzo will do

**Comment by keianhzo** [comment]
@daron-walters when exiting the volume button through the bottom side, the volume is not hidden. 

I think we could do something like:
- Show the volume slider when the hover enters the volume button 
- Schedule a hide whenever the hover exits the volume button or the volume slider
- Cancel the scheduled hide when the hover enters the volume slider.

I think that would simplify the logic and we can remove the widget level hover and the threshold related code.

**Comment by daron-walters** [comment]
@keianhzo I made the change where it is handled by the buttons and not the widget as a whole.  I will reach out tomorrow to make sure I full understand all scenarios before doing another pull request

**Comment by keianhzo** [comment]
@daron-walters Looks good, I think the only thing missing is to cancel the hide task when we enter the volume slider otherwise it gets dismissed while you are trying to change the volume. Squash the commit before pushing again so we can more clearly see the commit changes.

**Comment by daron-walters** [comment]
never did a squash on rebase before until now so let me know if it is correct.  Added the change Manuel informed me of as far as turning off timer when hover enters volume slider

**Comment by bluemarvin** [comment]
You should should only have one commit remaining after you squash your branch. Typically you would do this by running `git rebase -i origin/master` and then flattening all the commits but the first. But for some reason what ever you've done to your branch, it causes conflicts with it self when trying to flatten. I'm trying to figure out how to untangle it.

**Comment by bluemarvin** [comment]
I've flattened your commit and pushed it to this branch: https://github.com/MozillaReality/FirefoxReality/tree/v10/volume_slider_immersive_mode_flattened
Take a look. If it appears correct you can push it to this branch as follows:
`git push -f origin v10/volume_slider_immersive_mode_flattened:v10/volume_slider_immersive_mode`
Let me know if you have any questions.

**Comment by bluemarvin** [comment]
@daron-walters You unsquashed your commit. You need to start with the flattened branch that I created and cherry-pick your new commit on top of the flattened branch locally. Then push that to GitHub.

**Comment by daron-walters** [comment]
ok will do

**Pull request merged**